File: su/2026-04/23_boj_A와B2.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class Main {
    static String S, T;

    static int dfs(String txt) {
        if (txt.length() == S.length()) {
            return txt.equals(S) ? 1 : 0;
        }

        if (txt.charAt(txt.length() - 1) == 'A') {
            if (dfs(txt.substring(0, txt.length() - 1)) == 1) {
                return 1;
            }
        }

        if (txt.charAt(0) == 'B') {
            String next = new StringBuilder(txt.substring(1)).reverse().toString();
            if (dfs(next) == 1) {
                return 1;
            }
        }

        return 0;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        S = br.readLine().trim();
        T = br.readLine().trim();

        System.out.println(dfs(T));
    }
}

File: su/2026-04/23_boj_A와B2.py
S = input().strip()
T = input().strip()

# T에서 시작해서 S까지 도달하기 (역방향)

def dfs(txt):
    if len(txt) == len(S):
        return 1 if txt == S else 0

    if txt[-1] == 'A':
        if dfs(txt[:-1]):
            return 1

    if txt[0] == 'B':
        if dfs(txt[1:][::-1]):
            return 1

    return 0

print(dfs(T))